src/thread_pool.hpp

```diff
@@ -201,8 +186,8 @@ template <typename T, void (T::*process_func)(int index)>
 class nervana::thread_pool_queue
 {
 public:
-    thread_pool_queue(int thread_count)
-        : m_thread_pool(thread_count)
+    thread_pool_queue(std::vector<int> thread_affinity_map)
```

Comment:
Analogous comment as above - consider accepting reference or r-value reference to avoid copying.

Reply:
answer as above